openapi.yml

```diff
@@ -357,6 +357,25 @@ components:
       x-extensible-enum:
         - "LOCAL"
         - "NATIONAL"
+    RequestStatus:
+      description: |-
+        An enumerator of available statuses for a request.
+        Two different statuses of `CREATED` and `SUBMITTED` are available in order to map
+        the different steps for an organization registration and a user delegation, respectively:
+        1. the user creates the request with its related document
+        2. the user sends the document to the legal representative
```

Comment:
why we need to save a request on document creation ? (and not only when documents are sent)

Reply:
According to the onboarding flow, the delegate should check the documents after they are generated and they will be sent after he confirms their correctness, so two different calls to the server must be made.
If we don't save the request in the database on document creation, then its data should be sent again by the client. I think this is not the best solution, because the data sent by the client are already checked and validated in the first call and if they are sent again they should be checked and validated again. Moreover, the documents are already generated using the information provided in the first call, and there is no way to ensure that the data sent with the two calls are exactly the same.

Reply:
to recap, we need to save the request just after the documents are created AND in the moment they are sent, to keep a relationship between these signed documents we deliver to the user (as they refer some metadata relative to the registration procedure) and the following registration / delegation request.

in other words, if we don't store the data relative to the documents, the user may trigger a request relative to a document (now sent) which does not refer to the original metadata.